components/server/src/main/java/com/hotels/styx/server/track/CurrentRequestTracker.java

```diff
@@ -15,23 +15,19 @@
  */
 package com.hotels.styx.server.track;
 
+import com.hotels.styx.api.LiveHttpRequest;
+
 import java.util.Collection;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Supplier;
 
-import com.hotels.styx.api.LiveHttpRequest;
-
 /**
  * Manger class to manage the current requests.
  */
 public class CurrentRequestTracker implements RequestTracker {
-
-    private static ConcurrentHashMap<Object, CurrentRequest> currentRequests = new ConcurrentHashMap<>();
-
     public static final CurrentRequestTracker INSTANCE = new CurrentRequestTracker();
 
-    private CurrentRequestTracker() {
```

Comment:
Instead of letting the CurrentRequestTracker be the default, I would rather set the appropriate visibility (hopefully package private will suffice) and annotate the method with  @VisibleForTesting.  Probably we should redesign something if need public visibility as this is supposed to be a singleton...

Reply:
That's quite unfortunate.  I would still at least create a explicit constructor with the @VisibleForTesting annotation. Another option would be to have a private constructor and a static factory method that has the visible for testing annotation.